module_2/gifts/service/src/main/java/com/epam/esm/service/validation/CertificateValidation.java

```diff
@@ -0,0 +1,87 @@
+package com.epam.esm.service.validation;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.exception.ErrorCode;
+import com.epam.esm.exception.NullEntityException;
+import com.epam.esm.exception.ValidationException;
+import com.epam.esm.repository.CertificateRepository;
+import com.epam.esm.repository.model.CertificateModel;
+
+@Component
+public class CertificateValidation {
+	private static final int MIN_NAME_LENGTH = 5;
+	private static final int MIN_DESCRIPTION_LENGTH = 5;
+	private static final int MIN_DURATION = 1;
+	private static final int MAX_DURATION = 366;
+	private static final int MAX_NAME_LENGTH = 50;
+	private static final int MAX_DESCRIPTION_LENGTH = 1000;
+	private static final BigDecimal MAX_PRICE = new BigDecimal("99999.99");
+
+	@Autowired
+	private CertificateRepository certificateRepository;
+
+	public CertificateValidation() {
+
+	}
+
+	public void validateId(long id) {
+		if (id <= 0) {
+			throw new ValidationException("id = " + id, ErrorCode.INVALID_CERTIFICATE_ID);
+		}
+	}
+
+	public boolean validateName(String name) {
+		return Util.checkLength(name, MIN_NAME_LENGTH, MAX_NAME_LENGTH);
+	}
+
+	public boolean checkDublicatedName(String certificateName) {
+		if (!validateName(certificateName)) {
+			return false;
+		}
+		CertificateModel certificateModel = certificateRepository.readByCertificateName(Util.removeExtraSpaces(certificateName));
```

Comment:
@LizavetaSinitsyna Usually we use such method like `isExistCertificateByName` which return boolean value.